rcl/src/rcl/node.c

```diff
@@ -179,7 +211,47 @@ rcl_node_init(
   }
   // actual domain id
   node->impl->actual_domain_id = domain_id;
-  node->impl->rmw_node_handle = rmw_create_node(name, local_namespace_, domain_id);
+
+  const char * ros_enable_security = NULL;
+  const char * ros_enforce_security = NULL;
+
+  if (rcutils_get_env(sros_enable_security_var_name, &ros_enable_security)) {
+    RCL_SET_ERROR_MSG("fail fetching environment variable", rcl_get_default_allocator());
+    return RCL_RET_ERROR;
+  }
+
+  bool use_security = (0 == strcmp(ros_enable_security, "true"));
+
+  if (rcutils_get_env(sros_security_strategy_var_name, &ros_enforce_security)) {
+    RCL_SET_ERROR_MSG("fail fetching environment variable", rcl_get_default_allocator());
+    return RCL_RET_ERROR;
+  }
+
+  rmw_node_security_options_t node_security_options =
+    rmw_get_zero_initialized_node_security_options();
+  node_security_options.enforce_security = (0 == strcmp(ros_enforce_security, "Enforce")) ?
+    RMW_SECURITY_ENFORCEMENT_ENFORCE : RMW_SECURITY_ENFORCEMENT_PERMISSIVE;
+
+  if (!use_security) {
+    node_security_options.enforce_security = RMW_SECURITY_ENFORCEMENT_PERMISSIVE;
+    node_security_options.security_root_path = NULL;
```

Comment:
This is my (arguably dirty) way of making sure that the underlying rmw will not create a secure node if use_security is false. I invalidate the path in rcl rather than adding a field to the security_options structure (to say enable or disable) allowing to avoid duplicating the logic in each rmw_implementation